### ISSUE TITLE
Expand Functionality is not working on jpa processor for version 1.0.9

### ISSUE THREAD

**Issue by keshavsahu10** [SAP/olingo-jpa-processor-v4#218]
As we have upgraded jpa processor to 1.0.9, expand functionality is not working but same is working with 1.0.8

**Comment by keshavsahu10** [comment]
Odata Query : 
/**Workers**?$expand=**Title**
Both Worker and Title related to eachother by OneToOne,

Error : 
 getProcess : URL - /zip/odata/Workers , Query string - $expand=Title
2023-07-24 18:59:10.832  WARN 25528 --- [http-nio-8988-exec-2] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 169, SQLState: S0001
2023-07-24 18:59:10.834 ERROR 25528 --- [http-nio-8988-exec-2] o.h.engine.jdbc.spi.SqlExceptionHelper   : A column has been specified more than once in the order by list. Columns in the order by list must be unique.

**Comment by wog48** [comment]
One additional question:

If you look at generated SQL statement in the logs, does it also contain duplicates in the Select clause?

**Comment by keshavsahu10** [comment]
select
    distinct title1_.[LongDescription] as col_0_0_,
             title1_.[TitleId] as col_1_0_,
             title1_.[Code] as col_2_0_,
             title1_.[ShortDescription] as col_3_0_
from
    [tx].[vi_Worker] worker0_
        inner join
    [tx].[Title] title1_
    on worker0_.[TitleId]=title1_.[TitleId]
where
        worker0_.[WorkerId]>=1
  and worker0_.[WorkerId]<=1000
order by
    title1_.[TitleId] asc,
    title1_.[TitleId] asc;

**Comment by keshavsahu10** [comment]
In order by clause it's duplicate

**Comment by wog48** [comment]
Actually, I'm not able to reproduce this. The OrderBy clause is created in [JPAAbstractExpandQuery#createOrderByJoinCondition](https://github.com/SAP/olingo-jpa-processor-v4/blob/6da94c5f6a18900adba1577878e0310278cd8cbb/jpa/odata-jpa-processor/src/main/java/com/sap/olingo/jpa/processor/core/query/JPAAbstractExpandQuery.java#L99C12-L99C12 ). Looking at the code, I assume that there is  problem with the metadata generation. Is it okay to provide the code of entity _vi_Worker_?

**Comment by keshavsahu10** [comment]
Yes we can , but actually _vi_Worker_ is an **view** not **entity**, we are making use of view instead of entity as per our requirement, but same thing is working in 1.0.8 version.

**Comment by wog48** [comment]
It should not make a difference is your jpa entity based on a view or a table.
